src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import createFocusTrap from 'focus-trap';
+import { st, classes } from './Modal.st.css';
+import { TPAComponentsConsumer } from '../TPAComponentsConfig';
+import { ModalProps, ModalDefaultProps, ModalState } from './types';
+import { CloseIcon } from './Close';
+
+export class Modal extends React.Component<ModalProps, ModalState> {
+  static displayName = 'Modal';
+  static defaultProps: ModalDefaultProps = {
+    isOpen: false,
+    closeOnClickOutside: true,
+    minHeight: '150px',
+    maxHeight: '620px',
+    minWidth: 'auto',
+    maxWidth: '580px',
+    focusTrap: true,
+  };
+
+  el = null;
+  focusTrapInstance = null;
+
+  constructor(props: ModalProps) {
+    super(props);
+
+    this.el = document.createElement('div');
+    this.state = {
+      isCloseInProgress: false,
+    };
+  }
+
+  componentDidMount() {
+    this.props.rootElement.appendChild(this.el);
+    if (this.props.focusTrap) {
+      this.focusTrapInstance = createFocusTrap(this.el, {
+        escapeDeactivates: false,
+        clickOutsideDeactivates: false,
+        returnFocusOnDeactivate: false,
+      });
+    }
+  }
+
+  componentDidUpdate(prevProps: ModalProps) {
+    if (this.props.isOpen !== prevProps.isOpen && this.focusTrapInstance) {
+      if (this.props.isOpen) {
+        try {
+          this.focusTrapInstance.activate();
+        } catch (err) {
+          console.error(err);
+        }
```

Comment:
Is the try & catch really needed?
Maybe, instead of catching and printing an error, we can just remove the try & catch.
And if there will be errors, then will find them quicker this way

Reply:
Yep, focus trap plugin generate some extra errors is different cases, f.ex., if inside modal we haven't any focusable element. So I think try/catch is needed here.

Reply:
Oh I see.
So in that case, let's remove the `console.log`